torch/csrc/jit/serialization/pickler.cpp

```diff
@@ -801,4 +801,92 @@ bool checkHasValidSetGetState(const std::shared_ptr<c10::ClassType>& cls) {
   return true;
 }
 
+// Declare BackendMeta serialization and deserialization function pointer types.
+using BackendMetaPtr =
+    void (*)(const at::Tensor&, std::unordered_map<std::string, bool>&);
+
+// map to save function pointer for BackendMeta serialization.
+// key is the DeviceType, value is std::pair obj.
+// value.first represent get function and value.seconde represent set function
+static std::unordered_map<int, std::pair<void*, void*>> serialization_map;
```

Comment:
DeviceType is an enum, so I think you can store it directly, without conversion to int. Also, you have the function pointer type on line 805, no need to cast the function pointer to void*.